admin/class-yoast-notification-center.php

```diff
@@ -25,12 +25,18 @@ class Yoast_Notification_Center {
 	/** @var array Notifications that were resolved this execution */
 	private $resolved = 0;
 
+	/** @var array Internal storage for transaction before notifications have been retrieved from storage. */
+	private $queued_transactions = array();
+
+	/** @var bool Internal flag for whether notifications have been retrieved from storage. */
+	private $notifications_retrieved = false;
+
 	/**
 	 * Construct
 	 */
 	private function __construct() {
 
-		$this->retrieve_notifications_from_storage();
+		add_action( 'init', array( $this, 'setup_current_notifications' ), 1 );
```

Comment:
This also should be `admin_init`, as we don't display the notification numbers on the frontend.
This could have implications with notification counts being detected as changed, please do some testing to see if this happens.

Reply:
I think `init` works best as hook for the following reasons:
* It ensures the notifications are initialized as early as possible (the current user is set up right before `init` by core).
* Notification counts are shown in the admin bar, so `admin_init` wouldn't work because that also shows for frontend requests.